### ISSUE TITLE
an autograd kernel was not registered to the Autograd key(s) but we are trying to backprop through it.

### ISSUE THREAD

**Issue by pdhananjaya** [pytorch/xla#6758]
## 🐛 Bug

/usr/local/lib/python3.10/dist-packages/torch/autograd/__init__.py:266: UserWarning: aten::reshape: an autograd kernel was not registered to the Autograd key(s) but we are trying to backprop through it. This may lead to silently incorrect behavior. This behavior is deprecated and will be removed in a future version of PyTorch. If your operator is differentiable, please ensure you have registered an autograd kernel to the correct Autograd key (e.g. DispatchKey::Autograd, DispatchKey::CompositeImplicitAutograd). If your operator is not differentiable, or to squash this warning and use the previous behavior, please register torch::CppFunction::makeFallthrough() to DispatchKey::Autograd. (Triggered internally at ../torch/csrc/autograd/autograd_not_implemented_fallback.cpp:63.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass

## To Reproduce

I tried to run "The Annotated Diffusion Model" by HugginFace: https://huggingface.co/blog/annotated-diffusion
Following the "PYTORCH ON XLA DEVICES": https://pytorch.org/xla/release/2.2/index.html to train on single TPU, add xm.mark_step() after optimzer.step() to run on xla devices.

- In the training code in the colab after importing
   """
   import torch_xla.core.xla_model as xm
   device = xm.xla_device():
   """

- I modified the following to train the code on colab:
   """
   loss.backward()
   optimizer.step()
   """

- I replaced with 
   """
   loss.backward()
   optimizer.step()
   xm.mark_step()
   """

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]: TPU
 - torch_xla version: 2.2.0
 - torch version - 2.2.0

**Comment by JackCaoG** [comment]
This is a known issue, @wonjoolee95 is working on it.

**Comment by pdhananjaya** [comment]
Hi @JackCaoG, @wonjoolee95, any updates on the above mentioned issue ?
The issue still persists with modifying the training code for xla devices on Google Colab.

**Comment by wonjoolee95** [comment]
Apologies for the delay -- this has been a tricky one to solve. We're still working on a fix.

We've merged https://github.com/pytorch/xla/pull/6801 which should remove this warning message. However, we saw performance regression with this PR so we'll need to investigate a bit more.

On the other hand, this is just a warning message and should have no impact on the performance.
